koku/masu/database/aws_report_db_accessor.py

```diff
@@ -440,3 +440,62 @@ def check_for_matching_enabled_keys(self):
                 LOG.info(log_json(msg="no matching enabled keys for OCP on AWS", schema=self.schema))
                 return False
         return True
+
+    def delete_ec2_compute_summary_entries_for_date_range_raw(self, source_uuid, start_date, end_date):
+        """
+        Delete records from the EC2 compute summary table for a specified source and date range.
+
+        Args:
+            source_uuid (str): The unique identifier for the data source.
+            start_date (datetime.date): The start date of the range for which to delete records.
+            end_date (datetime.date): The end date of the range for which to delete records.
+
+        """
+
+        # date range should cover the full month
+        usage_start_date = start_date.replace(day=1)
+        table_name = self._table_map["ec2_compute_summary"]
+        msg = f"Deleting records from {table_name} for source {source_uuid} from {usage_start_date} to {end_date}"
+        LOG.info(msg)
+
+        sql = f"""
+            DELETE FROM {self.schema}.{table_name}
+            WHERE usage_start >= '{usage_start_date}'::date
+                AND usage_end <= '{end_date}'::date
+                AND source_uuid = '{source_uuid}'
+        """
+
+        self._execute_raw_sql_query(table_name, sql, operation="DELETE")
```

Comment:
Can we leverage the existing function of `delete_line_item_daily_summary_entries_for_date_range_raw` here? it looks like basically the same SQL and we can pass in the table and such: https://github.com/project-koku/koku/blob/main/koku/masu/database/report_db_accessor_base.py#L157